server.js

```diff
@@ -65,122 +61,161 @@ function reusableRequest(url, method){
+                            isCached = true;
+                            resolve(isCached);
+                        }else{
+                            isCached = false;
+                            resolve(isCached);
+                        }
+                });
+            });
+        })
+    }catch(error){
+        console.error(error);
+    }
+}
+
 //Returns promise once all api requests have finished
 function waitForDetails(userCoordinates){
-    return new Promise (function (resolve) {
-        responseObject = getResults(userCoordinates);
-        resolve(responseObject)
+    return new Promise (async function (resolve) {
+        let responseObject = await getResults(userCoordinates);
+        resolve(responseObject);
     })
 };
 //Awaits for API promise to resolve
 async function sendLocationResponse(req, res){
     try{
-        let responseObject = await waitForDetails(req.body.userCoordinates);
-        res.send(responseObject)
+        let responseObject = await waitForDetails(req.query.coordinates);
+        res.send(responseObject);
     }catch(error){
-        res.send(error)
+        res.send(error);
     }
 };
-
+//Returning results to the database
 function collectResults(placeDetailsObj){
-    MongoClient.connect(url, function(err, db) {
+    MongoClient.connect(dbConfig.url, function(err, db) {
         if (err) console.error(err);
-        let dbo = db.db(dbName);
-        let recentLocation = { 
-            name: placeDetailsObj.result.name, 
-            address: placeDetailsObj.result.formatted_address, 
-            googleMapsUrl: placeDetailsObj.result.url,
-            googleImageUrl: placeDetailsObj.result.photoUrls
-        };
-        dbo.collection("recent-locations").insertOne(recentLocation, function(err, res) {
-          if (err) console.error(err);
-          db.close();
+        let dbo = db.db(dbConfig.dbName);
+        dbo.collection("recentlocations").insertOne(placeDetailsObj, function(err, res) {
+            if (err) console.error(err);
+            db.close();
         });
-      });
+    });
 }
 
+async function sendRecentLocations(req, res){
+    try{
+        let result = await getRecentLocations();
+        jsonResponse = JSON.stringify(result);
+        res.send(jsonResponse);
+    }catch(error){
+        res.send(error);
+    };
+};
+
+function getRecentLocations(){
+    return new Promise (function (resolve) {
+        MongoClient.connect(dbConfig.url, function(err, db) {
+            if (err) console.error(err);
+            let dbo = db.db(dbConfig.dbName);
+            let mysort = { name: -1 };
+            dbo.collection("recentlocations").find().limit(3).sort(mysort).toArray(function(err, result) {
+              if (err) console.error(err);
+              db.close();
+              resolve(result);
+            });
+        });
+    });
+};
+
 /*
 API HANDLERS
 */
 
 //Handler for /go when userCoordinates is provided
 app.post('/newlocationsearch', (req, res) => {
```

Comment:
Just having a look at these handler functions here, they just call into another function, which doesn't really make sense. What might be better is to pull the try/catch block up a level, and then you can catch any errors and handle them, usually logging the error somewhere. You can also decide which response code to send back to the requester. It's probably not best practice to just return the error either, as we want to obsure as much of the back end from the FE as possible, maybe return a custom error object something like `{ "message":"There was an error, please try again", "status_code":500 }`

Reply:
Please check commit cced6f1dceea37573de7b01d2035013f24ed50e4, I've moved the try catch into the handler, which will now successfully respond based on responses from other APIs and any other request/DB errors. Also sends 500 with customer error json in the response.